client/src/modules/MentorRegistry/components/MentorRegistryTableContainer.tsx

```diff
@@ -0,0 +1,349 @@
+import { useMemo, useState } from 'react';
+import { GithubUserLink } from 'components/GithubUserLink';
+import { SafetyCertificateTwoTone } from '@ant-design/icons';
+import { colorTagRenderer, getColumnSearchProps, stringSorter, tagsRenderer, dateSorter } from 'components/Table';
+import { formatDate } from 'services/formatter';
+import { Course } from 'services/models';
+import CopyToClipboardButton from 'components/CopyToClipboardButton';
+import { MentorsRegistryColumnKey, MentorsRegistryColumnName, TABS, MentorRegistryTabsMode } from '../constants';
+import { FilterValue } from 'antd/lib/table/interface';
+import { Button } from 'antd';
+import { ColumnType } from 'antd/lib/table';
+import { DisciplineDto, MentorRegistryDto, UpdateUserDtoLanguagesEnum } from 'api';
+import css from 'styled-jsx/css';
+
+interface ChildrenProp {
+  tagFilters: string[];
+  filteredData: MentorRegistryDto[];
+  columns: ColumnType<MentorRegistryDto>[];
+  handleTagClose: (tag: string) => void;
+  handleClearAllButtonClick: () => void;
+  handleTableChange: (_: any, filters: Record<MentorsRegistryColumnKey, FilterValue | string[] | null>) => void;
+}
+
+interface Props {
+  mentors: MentorRegistryDto[];
+  courses: Course[];
+  activeTab: MentorRegistryTabsMode;
+  handleModalDataChange: (mode: string, record: MentorRegistryDto) => void;
+  children: (props: ChildrenProp) => JSX.Element;
+  disciplines: DisciplineDto[];
+}
+
+export interface CombinedFilter {
+  preselectedCourses: number[];
+  preferredCourses: number[];
+  technicalMentoring: string[];
+
+  filterTags?: string[];
+}
+
+export const MentorRegistryTableContainer = ({
+  children,
+  mentors,
+  courses,
+  activeTab,
+  handleModalDataChange,
+  disciplines,
+}: Props) => {
+  const [tagFilters, setTagFilters] = useState<string[]>([]);
+  const [combinedFilter, setCombinedFilter] = useState<CombinedFilter>({
+    preferredCourses: [],
+    preselectedCourses: [],
+    technicalMentoring: [],
+  });
+
+  const renderPreselectedCourses = (courses: Course[]) => {
+    return (values: number[], record: MentorRegistryDto) => {
+      return (
+        <>
+          {values
+            .map(v => ({
+              value: courses.find(c => c.id === v)?.name ?? v.toString(),
+              alias: courses.find(c => c.id === v)?.alias ?? '',
+              color: record.courses.includes(v) ? '#87d068' : undefined,
+            }))
+            .map(v => (v.color ? colorTagRenderer(v.value, v.color) : renderTagWithCopyButton(v.value, v.alias)))}
+        </>
+      );
```

Comment:
`React.Fragment` is unnecessary here :)  
```suggestion
      return (
        values
          .map(v => ({
            value: courses.find(c => c.id === v)?.name ?? v.toString(),
            alias: courses.find(c => c.id === v)?.alias ?? '',
            color: record.courses.includes(v) ? '#87d068' : undefined,
          }))
          .map(v => (v.color ? colorTagRenderer(v.value, v.color) : renderTagWithCopyButton(v.value, v.alias)))
      );
```